soroban-env-host/src/native_contract/token/allowance.rs

```diff
@@ -35,7 +35,11 @@ pub fn write_allowance(
     // validates the expiration and then returns the ledger seq
     // The expiration can be less than ledger seq if clearing an allowance
     let ledger_seq = e.with_ledger_info(|li| {
-        if expiration > li.sequence_number.saturating_add(li.max_entry_expiration) {
+        if expiration
+            > li.sequence_number
+                .saturating_sub(1)
```

Comment:
It's unfortunate that this logic is going to creep into contracts as well. I think we're going to see quite a few off by one errors (with or without your change).